app/models/reviewable_score.rb

```diff
@@ -11,13 +13,18 @@ class ReviewableScore < ActiveRecord::Base
   # To keep things simple the types correspond to `PostActionType` for backwards
   # compatibility, but we can add extra reasons for scores.
   def self.types
-    @types ||= PostActionType.flag_types.merge(PostActionType.score_types)
+    cached_score_types = Discourse.redis.get(REVIEWABLE_SCORE_TYPES_KEY)
```

Comment:
Would be nice for all these cache cases with a hash/json to have some helper, like:

```ruby
DiscourseRedis.get_set_json(KEY) do
   # this code is only reached if the cache is not set
   blah_find_object
end
```

Then the helper could handle the JSON serialize/deserialize and cache setting/getting part